bot/engine/src/main/kotlin/engine/config/BotDefinitionWrapper.kt

```diff
@@ -37,11 +38,20 @@ internal class BotDefinitionWrapper(val botDefinition: BotDefinition) : BotDefin
     @Volatile
     private var allStories: List<StoryDefinition> = botDefinition.stories
 
-    fun updateStories(configuredStories: List<ConfiguredStoryDefinition>) {
-        logger.debug { "refresh configured stories for ${botDefinition.botId}" }
+    private val dao: StoryDefinitionConfigurationDAO = injector.provide()
+
+    @Volatile
+    private var storyConfRedirections: Map<String, String?> = emptyMap()
+
+    fun updateStories(configuredStories: List<ConfiguredStoryDefinition>, applicationId: String?) {
         this.configuredStories = configuredStories.filter { it.answerType != builtin }.groupBy { it.id }
         //configured stories can override built-in
         allStories = (this.configuredStories + botDefinition.stories.groupBy { it.id }).values.flatten()
+        val storyConfigurations = with(botDefinition) {
```

Comment:
Why do you retrieve storyConfigurations from databases and do not use configuredStories param?